### PR TITLE
dev-util/mh: initial import

### PR DESCRIPTION
**Pull request by oz123**
Signed-off-by: Oz N Tiram <oz.tiram@gmail.com>

### CODE REVIEW

File: dev-util/mh/files/Makefile.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
+diff --git a/Makefile b/Makefile
+index 01023ca..f87ab56 100644
+--- a/Makefile
++++ b/Makefile
+@@ -14,6 +14,7 @@ endif
+ 
+ DEBUG = -g -fsanitize=address
+ CFLAGS = -Wall -lpcre2-8 #$(DEBUG)
```
Comment (SoapGentoo) [inline dev-util/mh/files/Makefile.patch:8]: 1. wrong, `-lpcre2-8` should be in `LIBS`
2. what about user `CFLAGS`?

File: dev-util/mh/files/Makefile.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
++LDFLAGS ?=
```
Comment (SoapGentoo) [inline dev-util/mh/files/Makefile.patch:9]: omit this definition here, adds no value

File: dev-util/mh/files/Makefile.patch [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ OPTS = -D COLOROUTPUT
+ PREFIX = /usr/local
+ MANPATH = /man/man
+@@ -27,7 +28,7 @@ clean:
+ 	rm -fv $(PROGNAME)
+ 
+ mh:  ## compile this software
+-	gcc -o $(PROGNAME) -D VERSION=\"$(VERSION)\" $(OBJECTS) $(CFLAGS) $(OPTS)
++	gcc -o $(PROGNAME) -D VERSION=\"$(VERSION)\" $(OBJECTS) $(CFLAGS) $(LDFLAGS) $(OPTS)
```
Comment (SoapGentoo) [inline dev-util/mh/files/Makefile.patch:18]: what if I want to use clang?

File: dev-util/mh/mh-1.0.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+EAPI=7
+
+DESCRIPTION="Generate help for Makefile"
+HOMEPAGE="https://github.com/oz123/mh"
+SRC_URI="https://github.com/oz123/mh/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="BSD"
+SLOT="0"
+IUSE="test"
+KEYWORDS="~amd64"
```
Comment (SoapGentoo) [inline dev-util/mh/mh-1.0.1.ebuild:12]: metadata block needs cleaning up

File: dev-util/mh/mh-1.0.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-libs/libpcre2"
```
Comment (SoapGentoo) [inline dev-util/mh/mh-1.0.1.ebuild:16]: so you're linking `libpcre2`, but not pulling it in a build time?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-util/mh/mh-1.0.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+BDEPEND="test? ( dev-util/cmocka )"
+
+PATCHES="${FILESDIR}/Makefile.patch"
+
+src_compile() {
+	emake mh VERSION="${PV}"
+}
+
+src_install() {
```
Comment (thesamesam) [inline dev-util/mh/mh-1.0.1.ebuild:26]: We can’t use your own install target?

is this because we need to fix the Makefile?

Comment (oz123) [reply dev-util/mh/mh-1.0.1.ebuild:26]: Yes, I think so. I can fix the makefile upstream and try again.
@SoapGentoo @thesamesam thanks for the review.

File: dev-util/mh/mh-1.0.1.ebuild (hunk 4 above)
Comment (oz123) [reply dev-util/mh/mh-1.0.1.ebuild:12]: Can you be specific? What should be removed?

File: dev-util/mh/mh-1.0.1.ebuild (hunk 5 above)
Comment (oz123) [reply dev-util/mh/mh-1.0.1.ebuild:16]: My bad. Will be fixed.

File: dev-util/mh/mh-1.0.1.ebuild (hunk 6 above)
Comment (oz123) [reply dev-util/mh/mh-1.0.1.ebuild:26]: @thesamesam using my own target, the binaries get installed to `/usr/local` ...

Comment (oz123) [reply dev-util/mh/mh-1.0.1.ebuild]: This is now fixed. I had to use PREFIX and DESTDIR (with some changes to the Makefile too).

File: dev-util/mh/files/Makefile.patch (hunk 2 above)
Comment (oz123) [reply dev-util/mh/files/Makefile.patch]: Removed.

File: dev-util/mh/files/Makefile.patch (hunk 1 above)
Comment (oz123) [reply dev-util/mh/files/Makefile.patch]: Fixed in the Makefile in version 1.0.3.

File: dev-util/mh/files/Makefile.patch (hunk 3 above)
Comment (oz123) [reply dev-util/mh/files/Makefile.patch]: Fixed, now the makefile uses $(CC) instead of  gcc.

File: dev-util/mh/mh-1.0.1.ebuild (hunk 4 above)
Comment (SoapGentoo) [reply dev-util/mh/mh-1.0.1.ebuild]: @oz123 I mistook this for another request to add a package to `::gentoo`. In general, we don't accept new packages to `::gentoo` anymore that are purely proxy maintained, for a number of reasons. Why not add it to `::guru`?

Comment (oz123) [reply dev-util/mh/mh-1.0.1.ebuild]: I can do that. But I don't contribute the guru project. That means I will need to go throuh the process of gaining access before this gets useful.
Quite a bummer. Out of scope here. But the. Direct engagement with Gentoo developers is really cool. Having another layer of abstraction is perhaps reducing the load on Gentoo developers but it's making it harder to recruit new developers.

**Comment by oz123** [comment]
added to guru.

**Pull request closed without merging**